pkg/util/socket.go

```diff
@@ -0,0 +1,32 @@
+package util
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// ErrUnixSocketNotExists is returned if the socket won't appear within the timeout duration.
+var ErrUnixSocketNotExists = errors.New("util/socket: unix socket does not exists")
+
+// WaitForUnixSocket waits for the duration of `timeout` until `path` exists by checking it every `interval`.
+// It returns `ErrUnixSocketNotExists` if the socket won't appear within the timeout, otherwise it returns nil.
+func WaitForUnixSocket(timeout time.Duration, interval time.Duration, path string) error {
```

Comment:
Nit: Would it be more efficient to use something like https://github.com/fsnotify/fsnotify to receive a notification from the OS that the file exists?

Reply:
I think it would, but since this is not in the data path, and we don't need low-latency, I think it would complicate this function as well. Do you think we need it?

Reply:
That's fair, no need for it now and we can think about it later if we discover it's something we want